apis/r/R/SOMAExperimentAxisQuery.R

```diff
@@ -1593,6 +1629,170 @@ SOMAExperimentAxisQuery <- R6::R6Class(
+          call. = FALSE
+        )
+      }
+
+      # Find the default layers
+      default_layers <- self$ms$X$get_metadata(default_hint)
+      if (!is.null(default_layers) && grepl(pattern = '^\\[', x = default_layers)) {
+        check_package('jsonlite')
+        default_layers <- jsonlite::fromJSON(default_layers)
+      }
+
+      # Initialize our dummy layer
+      counts <- list(Matrix::Matrix(
+        data = 0L,
+        nrow = length(features),
+        ncol = length(cells),
+        sparse = TRUE
+      ))
+      names(counts) <- lname
+      obj <- SeuratObject::.CreateStdAssay(
+        counts = counts,
+        cells = cells,
+        features = features,
+        transpose = FALSE
+      )
+
+      # Read in layers
+      for (lyr in layers) {
+        lyr_hint <- self$ms$X$get(lyr)$get_metadata(names(ragged_hint))
+        type_hint <- self$ms$X$get(lyr)$get_metadata(r_type_hint)
+        dnames <- self$ms$X$get(lyr)$dimnames()
+        attrn <- self$ms$X$get(lyr)$attrnames()
+        pkg <- NULL
+        if (!is.null(type_hint)) {
+          if (grepl(pattern = '^\\[', x = type_hint)) {
+            if (!requireNamespace('jsonlite', quietly = TRUE)) {
+              warning(warningCondition(
+                sprintf(
+                  "Layer '%s' is typed as '%s', but package 'jsonlite' is unavailable",
+                  lyr,
+                  type_hint
+                ),
+                class = "packageNotFoundWarning"
+              ))
+              next
+            }
+            type_hint <- jsonlite::fromJSON(type_hint)
+          } else if (grepl(pattern = s4_type, x = type_hint)) {
+            pkg <- strsplit(type_hint, split = ':')[[1L]][1L]
+            type_hint <- strsplit(type_hint, split = ':')[[1L]][2L]
+            if (!requireNamespace(pkg, quietly = TRUE)) {
+              warning(warningCondition(
+                sprintf(
+                  "Layer '%s' is typed as '%s:%s', but package '%s' is unavailable",
+                  lyr,
+                  pkg,
+                  type_hint,
+                  pkg
+                ),
+                class = "packageNotFoundWarning"
+              ))
+              next
+            }
+          }
+        }
+        lcells <- cells
+        lfeatures <- features
+        if (is.null(lyr_hint) || lyr_hint != ragged_hint[[1L]]) {
+          mat <- Matrix::t(self$to_sparse_matrix(
+            collection = 'X',
+            layer_name = lyr
+          ))
+        } else {
+          tbl <- tryCatch(
+            self$X(lyr)$tables()$concat(),
+            error = function(...) {
+              warning(warningCondition(
+                sprintf("Layer '%s' falls outside the query condition, skipping...", lyr),
+                class = "unqueryableLayerWarning"
+              ))
+              return(NULL)
+            }
+          )
+          if (is.null(tbl)) {
+            next
+          }
+          sdx <- vector("list", length = length(dnames))
```

Comment:
I have no idea what `sdx` means. Please use a longer name that indicates what this means.